src/main/java/org/opentripplanner/updater/vehicle_position/VehiclePositionPatternMatcher.java

```diff
@@ -249,6 +252,16 @@ else if (vehiclePosition.hasCurrentStopSequence()) {
     return newPosition.build();
   }
 
+  /**
+   * Checks that the current_stop_sequence can actually be found in the pattern.
+   */
+  private static boolean validStopSequence(
+    VehiclePosition vehiclePosition,
+    List<StopLocation> stopsOnVehicleTrip
+  ) {
+    return vehiclePosition.getCurrentStopSequence() < stopsOnVehicleTrip.size() - 1;
```

Comment:
I have two questions about this, can the stop sequence here be one that doesn't necessarily start at 0 index and can increase at random? This check wouldn't work in that case. Also, can the sequence be less than 0? What happens then?

Reply:
I think we spoke about this in the dev meeting. This and the original implementation assumes that the stop sequence in the GTFS-RT update starts at 0 and is consecutive but the GTFS spec doesn't require that.

The correct solution would be to look up the original GTFS stop sequence from the `TripTime` instance. I would like to do that in a follow-up.